vanilla-javascript-demo/index.html

```diff
@@ -0,0 +1,37 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>ChemDraw JS Quick Start</title>
+    <!-- Load the ChemDraw JS Library -->
+    <script src="asset/chemdrawweb/chemdrawweb.js"></script>
+  </head>
+  <body>
+    <!-- Create a div to attach the ChemDraw JS component to -->
+    <div
+      id="chemdrawjs-container" 
+      style="width: 640px; height: 480px;">
+    </div>
+    <script>
+      function onAttached(chemdraw) {
+        console.log("ChemDraw JS attached successfully");
+      }
+
+      function onAttachFailed(error) {
+        console.error("Could not attach ChemDraw JS", error.message);
+      }
+
+      fetch("asset/chemdraw-js-license.xml").then((res) => res.text()).then((license) => {
+        fetch("chemdraw-js.config.json").then(response => response.json()).then(config => {
+          // Attach ChemDraw JS to the container, providing our callbacks and the URL for the license file
+          perkinelmer.ChemdrawWebManager.attach({
+            id: "chemdrawjs-container",
+            callback: onAttached,
+            errorCallback: onAttachFailed,
+            license: license,
+            config: config
+          })
+        });
+      }).catch((e) => console.error(e));
```

Comment:
This isn't quite the right way to do this kind of async work. You will end up with something known as the pyramid of doom if you are not careful.

I suggest you rewrite this to use an self executing async function. Then you can await the fetches and avoid all the nesting.

Here's an example of what I mean...

```
(async function () {
  try {
    const licenseResponse = await fetch(...);
    const license = await licenseResponse.text();

    const configResponse = await fetch(...);
    const config = await configResponse.json();

    perkinelmer.ChemdrawWebManager.attach(...);
  } catch (error) {
    console.log(error);
  }
})();
```

Reply:
close pr to fix comments